examples/mnist/train.cpp

```diff
@@ -79,10 +79,12 @@ static void train_lenet(const std::string& data_dir_path) {
 
   progress_display disp(static_cast<unsigned long>(train_images.size()));
   timer t;
-  int minibatch_size = 10;
+  int minibatch_size = 16;
   int num_epochs     = 30;
 
-  optimizer.alpha *= static_cast<tiny_dnn::float_t>(std::sqrt(minibatch_size));
+  optimizer.alpha *=
+    std::min(tiny_dnn::float_t(4),
+             static_cast<tiny_dnn::float_t>(std::sqrt(minibatch_size)));
```

Comment:
@bhack The conclusions of the paper end with "We have demonstrated that Batch Normalization, while effective, is not well suited to small or non-i.i.d. training minibatches."

I tried to track down with blame what was the rationale for line 85 but maybe only the original author knows it. It seems mini-batch processing reduces the learning rate by O(sqrt(batch_size)) (I'll try to find the paper). I believe that's why that multiplying constant was added to it. I mentioned @Randl this fix is a hack to address a user that was having trouble with the code by playing with the mini-batch size.